ko/news/_posts/2016-11-09-ruby-2-4-0-preview3-released.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: news_post
+title: "루비 2.4.0-preview3 릴리스"
+author: "naruse"
+translator: "shia"
+date: 2016-11-09 18:00:00 +0900
+lang: ko
+---
+
+루비 2.4.0-preview3의 릴리스를 알리게 되어 기쁘게 생각합니다.
+
+루비 2.4.0-preview3은 루비 2.4.0의 세 번째 프리뷰입니다.
+이 프리뷰는 커뮤니티의 반응을 살펴보기 위해 릴리스되었습니다.
+기능에서 혼란을 느낀다면 부담없이 [피드백](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)을 보내주세요.
+
+## [Vladimir Makarov의 해시 테이블 개선을 도입](https://bugs.ruby-lang.org/issues/12142)
+
+해시 테이블(st_table)의 내부 구조를 삽입순 배열에 개방 주소법을 도입하는 개선이 이루어졌습니다.
+이 개선에 대해서는 많은 사람들의 토론이 있었으며, 특히 Yura Sokolov의 공헌이 컸습니다.
+
+## Binding#irb: `binding.pry`처럼 REPL 세션 시작하기
+
+디버깅 중에 `p`를 사용하여 변수의 값을 확인하는 경우가 잦습니다.
+[pry](https://github.com/pry/pry)를 사용하면 애플리케이션에 `binding.pry`를 추가하면 해당 위치에서 REPL을 실행해 줍니다.
+[r56624](https://github.com/ruby/ruby/commit/493e48897421d176a8faf0f0820323d79ecdf94a)는 irb에서도 `binding.irb`를 통해 같은 동작을 하는 기능을 추가합니다.
+
+## [Fixnum과 Bignum을 Integer로 통합](https://bugs.ruby-lang.org/issues/12005)
+
+[ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579)에서는
+Integer 클래스의 세부에 관해서 기술하고 있지 않아,
+루비에서는 2개의 정수 클래스(Fixnum과 Bignum)를 가지고 있습니다.
+루비 2.4에서는 이들이 Integer로 통합됩니다.
+Fixnum이나 Bignum에 영향을 주는 모든 C 확장이 변경되어야 합니다.
+
+[이 티켓](https://bugs.ruby-lang.org/issues/12005)과 [akr의 슬라이드](http://www.a-k-r.org/pub/2016-09-08-rubykaigi-unified-integer.pdf)를 참고하세요.
+
+## [String에서 유니코드 대/소문자 대응 지원](https://bugs.ruby-lang.org/issues/10085)
+
+`String/Symbol#upcase/downcase/swapcase/capitalize(!)`는 이제
+ASCII 대/소문자 대응 대신에 유니코드 대/소문자 대응을 지원합니다.
+
+## 성능 개선
+
+루비 2.4는 다음의 언어 변경을 포함한 성능 개선이 있습니다.
+
+### [Array#max, Array#min](https://bugs.ruby-lang.org/issues/12172)
+
+`[x, y].max`와 `[x, y].min`은 특정 조건에서 임시 배열을 생성하지 않도록
+최적화되었습니다.
+
+### [Regexp#match?](https://bugs.ruby-lang.org/issues/8110)
+
+`Regexp#match?`가 추가되었으며, 이는 후방 참조 객체를 생성하거나
+`$~`을 변경하지 않음으로써 객체 할당을 줄이며 정규 표현식을 매칭합니다.
+
+### 그 외의 성능 향상
+
+* [인스턴스 변수 접근 속도 향상](https://bugs.ruby-lang.org/issues/12274)
+
+## 디버깅
+
+### [Thread#report_on_exception과 Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+루비는 다른 스레드에서 명시적으로 접근하지 않는 이상 스레드에서의 예외를 무시합니다.
+
+### [Thread#report_on_exception과 Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+루비는 다른 스레드에서 명시적으로 접근하지 않는 이상 스레드에서의 예외를 무시합니다.
+`report_on_exception = true`를 사용하면,
+처리되지 않은 예외 때문에 스레드가 사망하는 경우, 알림을 받을 수 있게 됩니다.
+
+`report_on_exception`의 기본값으로 무엇을 사용하면 좋을지,
+그리고 스레드가 조인되지 않은 채 GC 되었을 때 알려주는 report-on-GC에 대한 피드백을 보내주세요.
+
+### [스레드 교착상태 탐지가 이제 백트레이스와 의존성 정보를 함께 보여주게 됩니다](https://bugs.ruby-lang.org/issues/8214)
+
+루비는 대기 중인 스레드의 교착상태 탐지를 지원합니다만,
+해당 리포트에서는 디버깅을 위한 충분한 정보가 포함되지 않았습니다.
+루비 2.4의 교착상태 탐지는 스레드의 백트레이스와 의존하고 있는 스레드에 대한 정보를 보여주게 됩니다.
+
+루비 2.4.0-preview3로 즐겁게 프로그램을 작성해보세요.
+그리고 여러분이 [느낀 점을 알려주세요](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)!
+
+## 2.3 이후의 주목할 만한 변경
+
+* OpenSSL 1.1.0을 지원합니다.
+* stdlib에서 ext/tk가 제거됩니다. [Feature #8539]
```

Comment:
It should be link. Isn't it? `[Feature #8539](https://bugs.ruby-lang.org/issues/8539)`